### DISCUSSION TITLE
MongoDB PUT request works locally, but not working when deployed.

### DISCUSSION THREAD

**Discussion by diceboi** [vercel/next.js#59096] [Help]
### Summary

Hello guys! Im really struggling with this for a couple days now. 

I have a live conenction to mongoDB, The integration was done, and works well with GET requests, and locally PUT requests as well. I try to create a simple admin dashboard to update some information in the database. The way it works: There is an input field with an edit button, and if I click to the edit button, the input field turns editable, and it can be saved with an other save button, that starts the PUT request. Just for reference:

`

const handleSave = async (index: number, id: string, newData: string, currentItem: any) => {
        try {
            const res = await fetch(`/api/updateData/${id}`, {
                method: 'PUT',
                headers: {
                    'Content-Type': 'application/json',
                },
                body: JSON.stringify({ newTitle: currentItem.title, newData }),
            });

            const responseJson = await res.json();

            if (!res.ok) {
                console.error('Failed to update data:', responseJson);
                throw new Error('Failed to update data');
            }

            console.log('Response from the server:', responseJson);
            
            // Log the data sent to the backend
            console.log('Data sent to backend:', { newTitle: currentItem.title, newData });
    
            // If the backend successfully updates the data, update the frontend state
            const updatedData = [...alapadatok];
            updatedData[index].data = newData; // Update the specific item's data

            setAlapadatok(updatedData);

            // Disable edit mode and reset the button after successful update
            const newEditMode = [...editMode];
            newEditMode[index] = false;
            setEditMode(newEditMode);
        } catch (error) {
            console.error('Error updating data:', error);
        }
    };

`

My PUT request at **api/updateData/[id]/route.ts:**

`

import Alapadatok from "@/modals/alapadatok";
import { NextResponse } from "next/server";

export async function PUT(request: any, { params }: any) {
  const { id } = params;
  const { newTitle:title , newData:data } = await request.json();

  try {
      const updatedDocument = await Alapadatok.findByIdAndUpdate(
        id,
          {
                "title": title,
                "data": data,
          },
      );

      if (!updatedDocument) {
        console.log('Nem található a dokumentáció');
          return NextResponse.json({ message: "Document not found" }, { status: 404 });
          
      }
      console.log('Frissítve');
      return NextResponse.json({ message: "Frissítve", updatedDocument }, { status: 200 });
  } catch (error) {
      console.error("Error updating data:", error);
      return NextResponse.json({ message: "Internal Server Error" }, { status: 500 });
  }
}

`

My modal at **modals/alapadatok.js:**

`

import mongoose from "mongoose";

const AlapadatokSchema = new mongoose.Schema(
  {
      title: String,
      data: String,
  },
  {
      timestamps: true,
  }
);

const Alapadatok = mongoose.models.Alapadatok || mongoose.model("Alapadatok", AlapadatokSchema);

export default Alapadatok;

`

My problem is: Its working fine locally, but when I deploy it, It always sends a 504 error.

**Browser console:**
page-46cc47dc9f83319d.js:1 
PUT https://admin-peboetterem-hu.vercel.app/api/updateData/65559f7a0b5e6dd72bbe2acb 504 (Gateway Timeout)
handleSave @ page-46cc47dc9f83319d.js:1
onClick @ page-46cc47dc9f83319d.js:1
Hi @ fd9d1056-2ef0b3d7324ff717.js:9
Ni @ fd9d1056-2ef0b3d7324ff717.js:9
Oi @ fd9d1056-2ef0b3d7324ff717.js:9
Nn @ fd9d1056-2ef0b3d7324ff717.js:9
Zm @ fd9d1056-2ef0b3d7324ff717.js:9
(anonymous) @ fd9d1056-2ef0b3d7324ff717.js:9
Ak @ fd9d1056-2ef0b3d7324ff717.js:9
ll @ fd9d1056-2ef0b3d7324ff717.js:9
zm @ fd9d1056-2ef0b3d7324ff717.js:9
xm @ fd9d1056-2ef0b3d7324ff717.js:9
wm @ fd9d1056-2ef0b3d7324ff717.js:9
472-218fb5ba84eb73e0.js:1 

Error updating data: Error: Failed to update data
at handleSave (page-46cc47dc9f83319d.js:1:1557)

**VSCode Console:**
Error updating data: MongooseError: Operation `alapadatoks.findOneAndUpdate()` buffering timed out after 10000ms
    at Timeout.<anonymous> (D:\Munkák\PEBo Étterem\rendelesek.peboetterem.hu\node_modules\mongoose\lib\drivers\node-mongodb-native\collection.js:186:23)
    at listOnTimeout (node:internal/timers:573:17)
    at process.processTimers (node:internal/timers:514:7)

I checked the database connection, its working fine.
Looked into mongodb whitelist, and the 0.0.0.0/0 is allowed.
Other than that, I dont know what to check. Can anyone help with that?

### Example

https://github.com/diceboi/admin.peboetterem.hu

**Comment by ekebobrown** [comment]
I am equally having a similar issue where method findById is working on my local setup but when uploaded on Vercel it doesn't anymore. Everything else endpoint works just fine

**Reply by tejashVaishnav** [reply]
have you found and work around?

**Comment by vishaldhangar15** [comment]
Anyone got the solution ?
